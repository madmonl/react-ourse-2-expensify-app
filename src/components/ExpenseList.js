import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import ExpensesSelector from '../selectors/expenses';

// Exporting both named(not connected to the store) version
// of Expense list, and default exported version - connected
// to the store, for actual running.
export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
        <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <div className="list-item list-item--message">
                        <span>No Expenses</span>
                    </div>
                    
                ) : (
                    props.expenses.map((expense) => (
                        <ExpenseListItem key={expense.id} {...expense}/>
                    ))
                )
            }
        </div>
    </div>
);

// filters is immutable as they are just props! 
// parent will see original filters
const mapStateToProps = (state) => {
    return {
        expenses: ExpensesSelector(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);
