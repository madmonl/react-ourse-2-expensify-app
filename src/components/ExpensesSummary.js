import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import expensesTotal from '../selectors/expenses-total';
import expensesSelector from '../selectors/expenses';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    
    return (
        <div>
            <h1> 
                Viewing {expenseCount} {expenseWord} totalling: {formattedExpensesTotal}.
            </h1>
        </div>
    );
};
    
const mapStateToProps = (state) => {
    const visibleExpenses = expensesSelector(state.expenses, state.filters);
    
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: expensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);