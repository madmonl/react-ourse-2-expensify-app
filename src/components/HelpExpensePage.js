import React from 'react';

const HelpExpensePage = () => (
    <div>
        This is from the help. 
    </div>
);

export default HelpExpensePage;