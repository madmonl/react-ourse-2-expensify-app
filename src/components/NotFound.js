import React from 'react';
import { Link } from 'react-router-dom';

// as opposed to the <a></a> tag that doesn't
// use client side refresh advantages.
// Link calls render instead of asking the server
// for the seekd html page.
const NotFound = () => (    
    <div>
        404! - <Link to="/">Go home.</Link>
    </div>
);

export default NotFound;
