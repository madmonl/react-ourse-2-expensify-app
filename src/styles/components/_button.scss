.button {
    color: white;
    background: $blue;
    border: none;
    // Our Links' margin and padding
    // are going to take into account.
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}