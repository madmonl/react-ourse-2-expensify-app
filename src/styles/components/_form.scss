// Targetting all direct children of form.
.form {
    display: flex;
    flex-direction: column;
    > * {
        margin-bottom: $m-size; 
    }
}

.form__error {
    font-style: italic;
}