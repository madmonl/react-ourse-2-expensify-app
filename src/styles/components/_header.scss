.header {
    background: $dark-blue;
}

.header__content {
    align-items: center;
    // puts content align
    display: flex;
    justify-content: space-between;
    padding: $s-size 0;
}

.header__title {
    color: white;
    // Removes the default decoration for Links 
    // which is an underline
    text-decoration: none;
    h1 {
        margin: 0;
    }
}