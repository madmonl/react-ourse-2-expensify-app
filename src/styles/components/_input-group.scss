.input-group {
    display: flex;
    flex-direction: column;
    margin-bottom: $m-size;

    @media (min-width: $desktop-breakpoint) {
        flex-direction: row; 
        margin-bottom: $m-size;
    }
}

.input-group__item {
    margin-bottom: $s-size;
    @media (min-width: $desktop-breakpoint) { 
        // Important!!
        // if margin-right: $s-size was the command,
        // we would have both margins on the bottom,
        // and on the right. 
        margin: 0 $s-size 0 0;
    }
}