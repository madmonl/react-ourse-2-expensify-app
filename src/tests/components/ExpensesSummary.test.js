import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';

test('Viewing 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={1003232230}/>)
    expect(wrapper).toMatchSnapshot();
});

test('Viewing 2 expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={13} expensesTotal={1000}/>)
    expect(wrapper).toMatchSnapshot();
});
