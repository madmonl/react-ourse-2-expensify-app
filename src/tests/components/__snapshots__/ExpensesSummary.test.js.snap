// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viewing 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling: 
      <span>
        $10,032,322.30
      </span>
      .
      <div
        className="page-header__actions"
      >
        <Link
          className="button"
          replace={false}
          to="/create"
        >
          Add Expense
        </Link>
      </div>
    </h1>
  </div>
</div>
`;

exports[`Viewing 2 expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        13
      </span>
       
      expenses
       totalling: 
      <span>
        $10.00
      </span>
      .
      <div
        className="page-header__actions"
      >
        <Link
          className="button"
          replace={false}
          to="/create"
        >
          Add Expense
        </Link>
      </div>
    </h1>
  </div>
</div>
`;
